### ISSUE TITLE
How collect gradients with two and more losses?

### ISSUE THREAD

**Issue by open-v** [tensorflow/tensorflow#36924]
```py
@tf.function
def train(dataset, epochs=5):
    for i in range(epochs):
        dataset = dataset.shuffle(1000)
        for images, boxes, labels in tqdm(dataset, desc='Epoch {} of {}'.format(i + 1, epochs)):
            with tf.GradientTape() as tape:
                localization, classification = model(images, training=True)
                loss_smooth = smooth_l1(boxes, localization)
                loss_focal  = focal(labels, classification)
            gradients = tape.gradient([loss_smooth, loss_focal], model.trainable_variables)
            optimizer.apply_gradients(zip(gradients, model.trainable_variables))
        print('Localization loss: {}, classification loss: {}'.format(loss_smooth, loss_focal))
```
How to can train model with two and more losses? Help?

**Comment by ravikyram** [comment]
@open-v 

Can you please go through the [link](https://www.pyimagesearch.com/2018/06/04/keras-multiple-outputs-and-multiple-losses/) and see if it helps you. If you still need support please fill [the Github new issue template](https://github.com/tensorflow/tensorflow/issues/new/choose). Please provide details about what platform you are using (operating system, architecture). Also include your TensorFlow version. Also, include simple standalone code to reproduce the issue in our environment. It helps us in localizing the issue faster. Thanks!

**Comment by luozhouyang** [comment]
You should combine the multi losses to one.

```python
total_loss = loss_smooth + loss_focal
```

or with a factor:

```python
total_loss = loss_smooth * 0.6 + loss_focal * 0.4
```

**Comment by open-v** [comment]
Thanks
